pandas/core/frame.py

```diff
@@ -8368,6 +8368,12 @@ def idxmin(self, axis=0, skipna=True) -> Series:
         """
         axis = self._get_axis_number(axis)
         indices = nanops.nanargmin(self.values, axis=axis, skipna=skipna)
+
+        # indices will always be np.ndarray since axis is not None and
+        # values is a 2d array for DataFrame
+        # error: Item "int" of "Union[int, Any]" has no attribute "__iter__"
+        indices = cast(np.ndarray, indices)
```

Comment:
What’s the reasoning here for cast instead of assert? The latter is generally easier to refactor

Reply:
#32785

Reply:
@WillAyd is this concern a blocker?